clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1885,6 +1886,38 @@ public void testUnsubscribeWithoutGroupId() {
         verify(applicationEventHandler).add(ArgumentMatchers.isA(UnsubscribeEvent.class));
     }
 
+    @Test
+    public void testSeekToBeginning() {
+        SubscriptionState subscriptions = mock(SubscriptionState.class);
+        TopicPartition topic = new TopicPartition("test", 0);
+        consumer = spy(newConsumer(
```

Comment:
Does `consumer` need to be wrapped as a `spy` in this case? It doesn't look like we're verifying or modifying it in any way 🤔

Reply:
You are right, we do not this, thanks.